### PR TITLE
Add `WsResponseBuilder` to build web socket session response

### LINKED ISSUE #1873: expose some way to alter websocket frame size limit

### PR DESCRIPTION
**Pull request by sazid**
## PR Type

Feature

## PR Checklist

- [x] Tests for the changes have been added / updated.
- [x] Documentation comments have been added / updated.
- [x] A changelog entry has been made for the appropriate packages.
- [x] Format code with the latest stable rustfmt
- [x] (Team) Label with affected crates and semver status.

## Overview

**New**: You can now start a websocket session with a frame size or a `actix_http::ws::Codec` by using the `ws:WsResponseBuilder`.

Example:

```rust
const MAX_FRAME_SIZE: usize = 10_000;
ws::WsResponseBuilder::new(WsActor, &req, stream)
    .codec(Codec::new()) // optional
    .protocols(&["A", "B"]) // optional
    .frame_size(MAX_FRAME_SIZE) // optional
    .start()
```

There are two terminal methods for the builder. This is in line with two existing `start()` and `start_with_addr()` fns.
- `start()` which starts the actor and returns a response.
- `start_with_addr()` which starts the actor and returns both the addr of the actor and the response.

Fixes #1873

### CODE REVIEW

File: actix-web-actors/CHANGES.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # Changes
 
 ## Unreleased - 2021-xx-xx
+* Add `ws:WsResponseBuilder` for building web socket session response.
```
Comment (robjtede) [inline actix-web-actors/CHANGES.md]:
```suggestion
* Add `ws:WsResponseBuilder` for building web socket session response. [#1920]

[#1920]: https://github.com/actix/actix-web/pull/1920
```

File: actix-web-actors/src/ws.rs [hunk 2]
```diff
@@ -29,6 +29,143 @@ use bytestring::ByteString;
...
+    codec: Option<Codec>,
+}
+
+impl<'a, A, T> WsResponseBuilder<'a, A, T>
+where
+    A: Actor<Context = WebsocketContext<A>>
+        + StreamHandler<Result<Message, ProtocolError>>,
+    T: Stream<Item = Result<Bytes, PayloadError>> + 'static,
+{
+    #[inline]
```
Comment (robjtede) [inline actix-web-actors/src/ws.rs]: none of these need to be inline

**Review by robjtede — CHANGES_REQUESTED** [review]

File: actix-web-actors/src/ws.rs (hunk 2 above)
Comment (sazid) [reply actix-web-actors/src/ws.rs]: Got it.

File: actix-web-actors/src/ws.rs [hunk 3]
```diff
@@ -279,6 +434,33 @@ where
         (addr, WebsocketContextFut::new(ctx, actor, mb, Codec::new()))
     }
 
+    #[inline]
+    /// Create a new Websocket context from a request, an actor, and a codec.
+    ///
+    /// Returns a pair, where the first item is an addr for the created actor,
+    /// and the second item is a stream intended to be set as part of the
+    /// response via `HttpResponseBuilder::streaming()`.
+    pub fn create_with_codec_addr<S>(
```
Comment (robjtede) [inline actix-web-actors/src/ws.rs]: is adding this new standalone fn necessary anymore?

**Review by robjtede — CHANGES_REQUESTED** [review]
i'd like to see one test where the frame size is exceeded producing an error

File: actix-web-actors/src/ws.rs (hunk 3 above)
Comment (sazid) [reply actix-web-actors/src/ws.rs]: I've moved it inside `WsResponseBuilder` since it's used only there internally. Is this what was expected?

**Comment by brockelmore** [comment]
Any update on this PR/when it will land? Seems stale, can fill in and complete any tasks that remain to be done

Edit: ping on this, definitively stale at this point, how can I step in to push it along?

**Review by robjtede — APPROVED** [review]

**Comment by robjtede** [comment]
thanks for getting this started @sazid and for everyone else's patience :)

**Pull request merged**